src/io.rs

```diff
@@ -2,85 +2,126 @@
 
 use core::{fmt, slice};
 use core::fmt::Write;
+use core::ptr;
 
 /// File descriptors
-const STDOUT: usize = 1;
-const STDERR: usize = 2;
+static mut STDOUT: isize = -1;
+static mut STDERR: isize = -1;
 
 /// Host's standard error
 struct Stderr;
 
 /// Host's standard output
 struct Stdout;
 
-fn write_all(fd: usize, mut buffer: &[u8]) {
+pub fn get_stdout() -> isize {
+    // Safe: 32-bit accesses are atomic on ARM
+    unsafe{ptr::read_volatile(&STDOUT)}
+}
+
+pub fn get_stderr() -> isize {
+    // Safe: 32-bit accesses are atomic on ARM
+    unsafe{ptr::read_volatile(&STDERR)}
+}
+
+/// Open stdout and stderr.
+pub fn open_streams() -> Result<(),()> {
```

Comment:
I see. But does that latch in the debugger? I'm thinking of something like this:

``` rust
fn main() {
    open_stdout();
    bkpt();
}
```

where the debug session goes like this:

``` console
> continue
10     bkpt();
> # stdout has been opened at this point

> monitor reset halt
> continue
10     bkpt();
> # stdout has been opened again
```

In this case you can't check whether stdout has been opened before within the same debug session. Storing the previous opened stdout fd (in RAM) doesn't help because that value is lost when you reset the microcotroller.